### PR TITLE
tests and doc for default registry

### PR DESCRIPTION
**Pull request by shubham-shay-te**

### CODE REVIEW

File: website/docs/default_registry.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+### Usage
+```python
+from core_lib.registry.default_registry import DefaultRegistry
+
+# Single Entry
+string = 'hello world'
+# initialize and provide data type to the registry
+registry_factory = DefaultRegistry(str)
```
Comment (shay-te) [inline website/docs/default_registry.md:78]: please store in the registry something more real other then just a string.

**Comment by shay-te** [comment]
good work thank you

**Review by shay-te — APPROVED** [review]

File: website/docs/default_registry.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        raise ValueError("register object is not of type \"{}\"".format(self._object_type))
+
+    if name in self.name_to_object:
+        raise ValueError("cache by name \"{}\" already registerd for type \"{}\"".format(name, object.__class__))
+
+    if is_default:
+        self.default_name = name
+    self.name_to_object[name] = object
+````
+`name` is type `str`, sets the name of the registry entry passed to the function.
```
[the remark this answers is not in the document]
Comment (shubham-shay-te) [reply website/docs/default_registry.md:26]: yes, `key` will make sense.

File: website/docs/default_registry.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+.
+registry_factory.registered()
+```
+
+### Example of DefaultRegistry
+```python
+from core_lib.registry.default_registry import DefaultRegistry
+
+# Single Entry
+user_name = 'Jon Doe'
```
Comment (shay-te) [inline website/docs/default_registry.md:118]: `DefaultRegistry` will be used to store more complex things and not strings. 
please take a look at `CacheRegistry` class. 
plase create a dummy class called `Customer`. 

+ show how to create the `CustomerRegistry`  take this example from the  CacheRegistry

File: tests/test_default_registry.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+        self.assertNotEqual(registry_factory.get(), dict_3)
+
+        self.assertEqual(registry_factory.registered()[0], 'dict_1')
+        self.assertEqual(registry_factory.registered()[1], 'dict_2')
+        self.assertEqual(registry_factory.registered()[2], 'dict_3')
+
+        self.assertListEqual(registry_factory.registered(), ['dict_1', 'dict_2', 'dict_3'])
+
+        registry_factory.unregister('dict_2')
+        self.assertEqual(registry_factory.get(), dict_1)
```
Comment (shay-te) [inline tests/test_default_registry.py:254]: same test for customer

File: tests/test_default_registry.py [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+        self.assertEqual(registry_factory.registered()[0], 'string')
+        self.assertListEqual(registry_factory.registered(), ['string'])
+
+        registry_factory.unregister('string')
+        self.assertEqual(registry_factory.get('string'), None)
+
+    def test_default_registry_int(self):
+        integer = 145
+        registry_factory = DefaultRegistry(int)
+        registry_factory.register('integer', integer)
```
Comment (shay-te) [inline tests/test_default_registry.py:42]: int_registry

File: website/docs/default_registry.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+
+Functions provided by the `DefaultRegistry` are as follows:
+
+- `register()`: Register's the key and value into the registry.
+````python
+def register(self, key: str, object, is_default: bool = False):
+    ...
+````
+`key` is type `str`, sets the key of the registry entry passed to the function.
```
Comment (shay-te) [inline website/docs/default_registry.md:36]: this key is unique. 
registering the same key again wioll raise an ValueError

**Review by shay-te — CHANGES_REQUESTED** [review]

**Review by shay-te — APPROVED** [review]

**Pull request merged**